systemtest/src/main/java/io/strimzi/systemtest/libClient/Consumer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.libClient;
+
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.IntPredicate;
+
+
+public class Consumer extends ClientHandlerBase<Integer> {
+    private static final Logger LOGGER = LogManager.getLogger(Consumer.class);
+    private Properties properties;
+    private final AtomicInteger numReceived = new AtomicInteger(0);
+    private final String topic;
+    private final String clientName;
+
+    Consumer(Properties properties, CompletableFuture<Integer> resultPromise, IntPredicate msgCntPredicate, String topic, String clientName) {
+        super(resultPromise, msgCntPredicate);
+        this.properties = properties;
+        this.topic = topic;
+        this.clientName = clientName;
+    }
+
+    @Override
+    protected void handleClient() {
+        KafkaConsumer<String, String> consumer = KafkaConsumer.create(vertx, properties);
+
+        if (msgCntPredicate.test(-1)) {
```

Comment:
Cause if `messageCount` is set to `-1` producer/consumer keep working till stop notification is received. It's not use din tests in current PR, but I want to use it in future PRs definitely. We will be able to produce/consumer messages during whole test execution.

Reply:
I understand the inner logic, so using the Vert.x service bus for verticle communication stopping the consumer/producer but because `handleClient` is called just one time when test starts I don't understand the check.
Or it can be called more times?

Reply:
I don't need event bus handler every time. That's the reason why I added there this check. In that case producer/consumer will not be stopped automatically by:
```
                    if (msgCntPredicate.test(numSent.get())) {
                        LOGGER.info("Producer produced {} messages", numSent.get());
                        resultPromise.complete(numSent.get());
                    }```
but it will wait for notification. Does it make sense now? Maybe it's not 100% clear approach since it's a little bit confusing